stress_test_ceph_vmware/cephops.py

```diff
@@ -49,21 +49,21 @@ def unset_noup(self):
 
     def get_down_osds(self):
         data, _ = self.client.run_cmd('ceph osd tree -f json')
-        down_osds = []
+        self.down_osds = []
```

Comment:
This is not neccessary. See L20: if this file

The constructor of this class is calling 

``` python

       self.down_osds = self.get_down_osds()
       self.out_osds = self.get_out_osds()
```

which makes the assignment of `self.down_osds` redundant.

If you encountered that `down_osds` is not populated you might the root cause in this function.

Reply:
the issue that I saw was that the `self.out_osds` was populated initially through the constructor but when the script is marking some disk out it doesn't get updated and that causes removing disks ignoring the threshold which in the end breaks the ceph cluster. I first checked the `get_out_osds()` function and it looks good.